core/services/llo/data_source.go

```diff
@@ -147,26 +150,36 @@ func (d *dataSource) Observe(ctx context.Context, streamValues llo.StreamValues,
 
 	wg.Wait()
 
+	elapsed := time.Since(now)
+
 	// Failed observations are always logged at warn level
+	var successfulStreamIDs []streams.StreamID
 	var failedStreamIDs []streams.StreamID
-	if len(errs) > 0 {
+	var errStrs []string
+
+	// Only log errors or if VerboseLogging is turned on
+	if len(errs) > 0 || opts.VerboseLogging() {
+		successfulStreamIDs = make([]streams.StreamID, 0, len(streamValues))
+		for strmID := range streamValues {
+			successfulStreamIDs = append(successfulStreamIDs, strmID)
+		}
+		sort.Slice(successfulStreamIDs, func(i, j int) bool { return successfulStreamIDs[i] < successfulStreamIDs[j] })
```

Comment:
Why do we need to log successful StreamIDs if we're logging the ones that failed underneath?

Reply:
If VerboseLogging is turned on, we log successful/failed even if nothing failed. If VerboseLogging is turned off, we only log if there was at least one error.

Reply:
How about `slices.Sort` since you are using the natural order?
```suggestion
		slices.Sort(successfulStreamIDs)
```